js/apps/admin-ui/src/util.ts

```diff
@@ -171,3 +171,18 @@ export const addTrailingSlash = (url: string) =>
   url.endsWith("/") ? url : url + "/";
 
 export const generateId = () => Math.floor(Math.random() * 1000);
+
+type PathParam = { [key: string]: string | null };
```

Comment:
This doesn't have to accept `null` values as we never pass anything but `string` into this function.
```suggestion
type PathParam = { [key: string]: string };
```
Or

```suggestion
type PathParam = Record<string, string>;
```